dali/pipeline/operators/reader/nvdecoder/imgproc.cu

```diff
@@ -24,139 +24,171 @@ namespace {
-    yuv.u = uv.x;
-    yuv.v = uv.y;
-
-    auto* out = &dst[(dst_x + dst_y * dst_width) * c];
-
-    size_t stride = 1;
-    yuv2rgb<float, float, Normalized>(yuv, out, stride);
+  cudaTextureObject_t luma, cudaTextureObject_t chroma,
+  T* dst, int index,
+  float fx, float fy,
+  int dst_width, int dst_height, int c) {
+  const int dst_x = blockIdx.x * blockDim.x + threadIdx.x;
+  const int dst_y = blockIdx.y * blockDim.y + threadIdx.y;
+
+  if (dst_x >= dst_width || dst_y >= dst_height)
+      return;
+
+  auto src_x = 0.0f;
+  src_x = static_cast<float>(dst_x) * fx;
+  auto src_y = static_cast<float>(dst_y) * fy;
+
+  // TODO(spanev) something less hacky here, why 4:2:0 fails on this edge?
+  float shift = (dst_x == dst_width - 1) ? 0 : 0.5f;
+
+  // https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#tex2d-object
+  yuv<float> yuv;
+  yuv.y = tex2D<float>(luma, src_x + shift, src_y + shift);
+  auto uv = tex2D<float2>(chroma, (src_x / 2) + shift, (src_y / 2) + shift);
+  yuv.u = uv.x;
+  yuv.v = uv.y;
+
+  auto* out = &dst[(dst_x + dst_y * dst_width) * c];
+
+  constexpr size_t stride = 1;
+  if (RGB) {
+    yuv2rgb<float, T, Normalized>(yuv, out, stride);
+  } else {
+    constexpr float scaling = Normalized ? 1.0f : 255.0f;
+    out[0] = yuv.y * scaling;
+    out[stride] = yuv.u * scaling;
+    out[stride*2] = yuv.v * scaling;
+  }
 }
 
 inline constexpr int divUp(int total, int grain) {
-    return (total + grain - 1) / grain;
+  return (total + grain - 1) / grain;
 }
 
 }  // namespace
 
 template<typename T>
 void process_frame(
-    cudaTextureObject_t chroma, cudaTextureObject_t luma,
-    SequenceWrapper& output, int index, cudaStream_t stream,
-    uint16_t input_width, uint16_t input_height) {
-    auto scale_width = input_width;
-    auto scale_height = input_height;
-
-    auto fx = static_cast<float>(input_width) / scale_width;
-    auto fy = static_cast<float>(input_height) / scale_height;
-
-    dim3 block(32, 8);
-    dim3 grid(divUp(output.width, block.x), divUp(output.height, block.y));
-
-    int frame_stride = index * output.height * output.width * output.channels;
-    LOG_LINE << "Processing frame " << index
-             << " (frame_stride=" << frame_stride << ")" << std::endl;
-    auto* tensor_out = output.sequence.mutable_data<T>() + frame_stride;
-
-    // TODO(spanev) Handle normalized and YCbCr versions
-    constexpr bool normalized = false;
-    process_frame_kernel<T, normalized><<<grid, block, 0, stream>>>
-            (luma, chroma, tensor_out, index, fx, fy, output.width, output.height, output.channels);
+  cudaTextureObject_t chroma, cudaTextureObject_t luma,
+  SequenceWrapper& output, int index, cudaStream_t stream,
+  uint16_t input_width, uint16_t input_height,
+  bool rgb, bool normalized) {
+  auto scale_width = input_width;
+  auto scale_height = input_height;
+
+  auto fx = static_cast<float>(input_width) / scale_width;
+  auto fy = static_cast<float>(input_height) / scale_height;
```

Comment:
This now useless will soon be replaced, as we might add in-Reader rescale support.
Please ignore the useless calculation here